lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -41,6 +48,30 @@ def process_root_folder
                 end
             end
 
+            # Manages folder free space
+            #
+            # The method will check if there is enough space to save more log files
+            # according to pre-established threshold.
+            #
+            # @param [integer] free_space_low_limit: required free space threshold, at
+            #   which the archiver starts deleting the oldest log files
+            # @param [integer] free_space_delete_until: post-deletion free space, at which
+            #   the archiver stops deleting the oldest log fil
+            def ensure_free_space(free_space_low_limit: @free_space_low_limit,
+                    free_space_delete_until: @free_space_delete_until)
+                stat = Sys::Filesystem.stat(@target_dir)
+                free_space = stat.bytes_free
+
+                return if free_space > free_space_low_limit
+
+                candidates = each_file_from_path(path).to_a.sort
+                loop do |index|
```

Comment:
```
loop do
    ....
    break if <condition>
end
```

is a while. Ruby has both `while` and `until`. I would have personally preferred

```
until free_space >= free_space_delete_until
end
```